### PR TITLE
Added business logic to sign up page

### PR DESCRIPTION
**Pull request by RomanDeveloperAcc**
Added business logic to sign up page including:
-new back-end functionality
-markup and style correctives

### CODE REVIEW

File: src/app/component/auth/new-sign-up/new-sign-up.component.ts [hunk 1]
```diff
@@ -9,13 +18,135 @@ import { MatDialogRef } from '@angular/material/dialog';
...
+    this.emailErrorMessageBackEnd = null;
+    this.passwordErrorMessageBackEnd = null;
+    this.passwordConfirmErrorMessageBackEnd = null;
+  }
+
+  private onSubmit(userOwnRegister: UserOwnSignUp): void {
+    this.setNullAllMessage();
+    this.loadingAnim = true;
+    this.userOwnSecurityService.signUp(userOwnRegister).subscribe(
+      () => {
```
Comment (nabermi2020) [inline src/app/component/auth/new-sign-up/new-sign-up.component.ts:56]: Please rewrite it(subscribe(this.onSubmitSuccess.bind(this), ...)

File: src/app/component/auth/new-sign-up/new-sign-up.component.ts [hunk 2]
```diff
@@ -9,13 +18,135 @@ import { MatDialogRef } from '@angular/material/dialog';
...
+          this.emailErrorMessageBackEnd = error.message;
+          break;
+        case 'password' :
+          this.passwordErrorMessageBackEnd = error.message;
+          break;
+        case 'passwordConfirm' :
+          this.passwordConfirmErrorMessageBackEnd = error.message;
+          break;
+      }
+    });
```
Comment (nabermi2020) [inline src/app/component/auth/new-sign-up/new-sign-up.component.ts:87]: Please add empty line

File: src/app/component/auth/new-sign-up/new-sign-up.component.ts [hunk 3]
```diff
@@ -9,13 +18,135 @@ import { MatDialogRef } from '@angular/material/dialog';
...
+    this.loadingAnim = false;
+  }
+
+  private signInWithGoogle(): void {
+    this.authService.signIn(GoogleLoginProvider.PROVIDER_ID).then(data => {
+      this.googleService.signIn(data.idToken).subscribe(
+        (data1: UserSuccessSignIn) => {
+          this.signInWithGoogleSuccess(data1);
```
Comment (nabermi2020) [inline src/app/component/auth/new-sign-up/new-sign-up.component.ts:95]: The same as line 56

File: src/app/component/auth/new-sign-up/new-sign-up.component.ts [hunk 4]
```diff
@@ -9,13 +18,135 @@ import { MatDialogRef } from '@angular/material/dialog';
...
+
+  private matchPassword(passInput: HTMLInputElement,
+                        passRepeat: HTMLInputElement,
+                        inputBlock: HTMLElement): void {
+    inputBlock.className = passInput.value !== passRepeat.value ?
+                           'main-data-input-password wrong-input' :
+                           'main-data-input-password';
+  }
+
+  private hideShowPassword(htmlInput: HTMLInputElement,
```
Comment (nabermi2020) [inline src/app/component/auth/new-sign-up/new-sign-up.component.ts:135]: Please rename it

File: src/app/component/auth/new-sign-up/new-sign-up.component.html [hunk 5]
```diff
@@ -19,86 +19,185 @@
     <div class="main-content-container">
       <div class="title">
         <h1 class=""title-text>
-          Вітаємо!
+          {{ 'user.auth.sign-up.title-up' | translate }}
```
Comment (nabermi2020) [inline src/app/component/auth/new-sign-up/new-sign-up.component.html:22]: Please fix it
image

Comment (RomanDeveloperAcc) [reply src/app/component/auth/new-sign-up/new-sign-up.component.html:22]: Fixed

File: src/app/component/auth/new-sign-up/new-sign-up.component.ts (hunk 2 above)
Comment (RomanDeveloperAcc) [reply src/app/component/auth/new-sign-up/new-sign-up.component.ts:84]: Fixed

File: src/app/component/auth/new-sign-up/new-sign-up.component.ts (hunk 1 above)
Comment (RomanDeveloperAcc) [reply src/app/component/auth/new-sign-up/new-sign-up.component.ts]: Fixed

File: src/app/component/auth/new-sign-up/new-sign-up.component.ts (hunk 3 above)
Comment (RomanDeveloperAcc) [reply src/app/component/auth/new-sign-up/new-sign-up.component.ts]: Fixed

File: src/app/component/auth/new-sign-up/new-sign-up.component.ts (hunk 4 above)
Comment (RomanDeveloperAcc) [reply src/app/component/auth/new-sign-up/new-sign-up.component.ts]: Fixed

File: src/app/component/auth/new-sign-up/new-sign-up.component.html [hunk 6]
```diff
@@ -19,86 +19,185 @@
...
+                 'wrong-input':'main-data-input'"
+                 required
+                 minlength="1"
+                 email="true"
           >
-          <label class="content-label">
-            Ваше ім’я
+          <div class="error-message"
+               [ngClass]="emailControl.invalid &&
+                            (emailControl.dirty||emailControl.touched)?
```
Comment (Gusvarov) [inline src/app/component/auth/new-sign-up/new-sign-up.component.html:52]: add spaces before and after ||

File: src/app/component/auth/new-sign-up/new-sign-up.component.html [hunk 7]
```diff
@@ -19,86 +19,185 @@
...
+                 maxlength="20"
           >
           <label class="content-label">
-            Пароль
+            {{ 'user.auth.sign-up.password' | translate }}
           </label>
-          <div class="main-data-input-password">
+          <div class="main-data-input-password"
+               [ngClass]="passwordControl.invalid && (passwordControl.dirty||passwordControl.touched) ?
+               'main-data-input-password wrong-input':'main-data-input-password'"
```
Comment (Gusvarov) [inline src/app/component/auth/new-sign-up/new-sign-up.component.html:82]: same as line 52

File: src/app/component/auth/new-sign-up/new-sign-up.component.html [hunk 8]
```diff
@@ -19,86 +19,185 @@
...
+                   #inputPasswordImg
+                   (click)="setPasswordVisibility(inputPassword, inputPasswordImg)"
               >
             </span>
           </div>
-          <label class="content-label">
-            Повторіть пароль
+          <div class="error-message"
+               [ngClass]="passwordControl.invalid && (passwordControl.dirty||passwordControl.touched) ?
+               'error-message-show' : 'error-message'"
```
Comment (Gusvarov) [inline src/app/component/auth/new-sign-up/new-sign-up.component.html:105]: same as line 52

File: src/app/component/auth/new-sign-up/new-sign-up.component.html [hunk 9]
```diff
@@ -19,86 +19,185 @@
...
                    class="show-password-img"
+                   #repeatPasswordImg
+                   (click)="setPasswordVisibility(repeatPassword, repeatPasswordImg)"
               >
             </span>
           </div>
+          <div class="error-message"
+               [ngClass]="passwordControlConfirm.invalid &&
+               (passwordControlConfirm.dirty||passwordControlConfirm.touched) ||
+               (inputPassword.value !== repeatPassword.value && repeatPassword.value !== '') ?
```
Comment (Gusvarov) [inline src/app/component/auth/new-sign-up/new-sign-up.component.html:148]: same as line 52

File: src/app/component/auth/new-sign-up/new-sign-up.component.html (hunk 8 above)
Comment (RomanDeveloperAcc) [reply src/app/component/auth/new-sign-up/new-sign-up.component.html:107]: fixed

File: src/app/component/auth/new-sign-up/new-sign-up.component.html (hunk 9 above)
Comment (RomanDeveloperAcc) [reply src/app/component/auth/new-sign-up/new-sign-up.component.html:150]: fixed

File: src/app/component/auth/new-sign-up/new-sign-up.component.html (hunk 6 above)
Comment (RomanDeveloperAcc) [reply src/app/component/auth/new-sign-up/new-sign-up.component.html]: fixed

File: src/app/component/auth/new-sign-up/new-sign-up.component.html (hunk 7 above)
Comment (RomanDeveloperAcc) [reply src/app/component/auth/new-sign-up/new-sign-up.component.html]: fixed

**Pull request merged**